fastos/src/vespa/fastos/unix_thread.cpp

```diff
@@ -61,6 +63,13 @@ bool FastOS_UNIX_Thread::Initialize (int stackSize, int stackGuardSize)
 
     pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
 
+    size_t adjusted_stack_size = stackSize;
+#ifdef __linux__
+    ssize_t stack_needed_by_system = __pthread_get_minstack(&attr);
+    adjusted_stack_size += stack_needed_by_system;
```

Comment:
The adjusted stack size might be slightly too large since PTHREAD_STACK_MIN is included in stack_needed_by_system.